### PR TITLE
[REVIEW] Add offsets for the categorical columns

### PR DESCRIPTION
**Pull request by oyilmaz-nvidia**
This is the PR to fix the bug reported in this issue; https://github.com/NVIDIA/NVTabular/issues/707

### CODE REVIEW

File: nvtabular/inference/triton/__init__.py [hunk 1]
```diff
@@ -463,6 +472,13 @@ def get_column_types(path):
     return json.load(open(os.path.join(path, "column_types.json")))
 
 
+def get_slot_sizes(path):
+    if not path.exists(path):
+        return json.load(open(os.path.join(path, "slot_size_array.json")))
```
Comment (benfred) [inline nvtabular/inference/triton/__init__.py:477]: Why are you checking if the path doesn't exist? Also if the path doesn't exist, how can the slot_size_array.json file inside the path exist?

File: nvtabular/inference/triton/__init__.py [hunk 2]
```diff
@@ -463,6 +472,13 @@ def get_column_types(path):
...
+    else:
+        return None
```
Comment (benfred) [inline nvtabular/inference/triton/__init__.py:479]: The only caller of this function seems to be in model_hugectr.py - maybe we should move there?

Also the code in model_hugectr.py doesn't handle this function returning None - maybe just remove the check to see if the file exists?

File: nvtabular/inference/triton/model_hugectr.py [hunk 3]
```diff
@@ -51,6 +51,9 @@ def initialize(self, args):
         self.workflow = nvtabular.Workflow.load(workflow_path)
         self.model_config = json.loads(args["model_config"])
         self.column_types = get_column_types(workflow_path)
+        self.slot_sizes = get_slot_sizes(workflow_path)
+        self.slot_sizes.insert(0, 0)
+        self.slot_sizes.pop()
```
Comment (benfred) [inline nvtabular/inference/triton/model_hugectr.py:56]: Does this work? The slot sizes file is written out as a dictionary afaict on line 138 of ```__init__.py``` , and this code seems to be expecting a list

**Review by benfred — CHANGES_REQUESTED** [review]
You need to run black on the changes here - the style check is failing because of it.

How have you tested this code? I've left some comments but I'm very doubtful that this PR works.

File: nvtabular/inference/triton/model_hugectr.py (hunk 3 above)
Comment (oyilmaz-nvidia) [reply nvtabular/inference/triton/model_hugectr.py]: Forgot to push some of the updates before turning the title to "REVIEW". Just pushed them.

File: nvtabular/inference/triton/__init__.py (hunk 2 above)
Comment (oyilmaz-nvidia) [reply nvtabular/inference/triton/__init__.py]: Updated this. And, moved it into model_hugectr.py

File: nvtabular/inference/triton/__init__.py (hunk 1 above)
Comment (oyilmaz-nvidia) [reply nvtabular/inference/triton/__init__.py]: Updated it.

**Comment by benfred** [comment]
rerun tests

**Comment by shashank3959** [comment]
Onur and I tested the change, and it seemed to work. 

Using the merlin v0.4 inference container, we observed an issue loading the ensemble model with the following setting in dcn_nvt/config.pbtxt
instance_group {
  count: 1
  kind: KIND_CPU
}


After removing this config, the ensemble loaded successfully.

**Comment by oyilmaz-nvidia** [comment]
@shashank3959 We have created an issue here to add a unit test to check the accuracy. Here is that issue; https://github.com/NVIDIA/NVTabular/issues/710

We'll add that test in the next release.

**Review by benfred — APPROVED** [review]

**Comment by benfred** [comment]
rerun tests

**Comment by vinhngx** [comment]
Great work. Thanks for promptly fixing this.

I've got a question. In the following:
output_df[self.column_types["cats"]] + self.slot_sizes

is slot_sizes the actual slot sizes, or accumulative sum of the slot sizes?

I think it needs to be the accumulative sum.

**Comment by oyilmaz-nvidia** [comment]
@vinhngx Created another PR for the accumulative sum. Could you please review this? 
https://github.com/NVIDIA/NVTabular/pull/735

Will add you as a reviewer.

**Pull request merged**